apps/root/src/services/contractService.ts

```diff
@@ -48,22 +49,33 @@ export default class ContractService {
 
   // ADDRESSES
   getHUBAddress(chainId: number, version?: PositionVersions): Address {
-    return (
-      HUB_ADDRESS[version || LATEST_VERSION][chainId] || HUB_ADDRESS[LATEST_VERSION][chainId]
-    ).toLowerCase() as Address;
+    const versionToUse = version || LATEST_VERSION;
+
+    const hubAddress =
```

Comment:
I dont think this should return something if it does not exist on the chain that we asked for, we should instead handle on parseLogs that this can come up as undefined